include/aspect/geometry_model/ellipsoidal_chunk.h

```diff
@@ -38,12 +40,77 @@ namespace aspect
     using namespace dealii;
 
     /**
+     * Define a type to know what type of topography we
+     * are using
+     */
+    enum topo_types {NO_TOPOGRAPHY,
+                     PRM_EXACT,
+                     PRM_UNIFORM_GRID_INTERPOLATED,
+                     FILE_UNIFORM_GRID,
+                     FILE_NONUNIFORM_GRID
+                    };
+
+    /**
      * A class that describes a geometry for an ellipsoid such as the WGS84 model of the earth.
      */
     template <int dim>
     class EllipsoidalChunk : public Interface<dim>
     {
       public:
+        static bool In2dPolygon(dealii::Point<2> &point,const std::vector<std::vector<double> > &pointList);
+        /**
+         * A class which describes the topography
+         */
+        class EllipsoidalChunkTopography
+        {
+          public:
+            EllipsoidalChunkTopography ();
+
+            /**
+             * Retrieve topography value at point lon,lat
+             */
+            double value (const double lon,
+                          const double lat) const;
+
+            /**
+             * The set functions
+             */
+            void set_corners(std::vector<Point<2> > corners);
+            void set_topography_type(topo_types topo_type);
+            void set_point_lists(std::vector<std::vector<std::vector<double> > > point_lists);
+            void set_topography_values (std::vector<double> topography_values);
+            void set_topography_data (Function<2> *topography_data);
+            void set_grid_number_data_points(std::vector<double> &grid_number_data_points);
+            void set_topography_file(std::string topo_file_name);
+
+            /**
+             * The get functions
+             */
+            topo_types                                   get_topo_type () const;
+            std_cxx11::array<std::pair<double,double>,2> get_endpoints () const;
+            std_cxx11::array<unsigned int,2>             get_number_of_intervals () const;
+            std::vector<double>                          get_data () const;
+            std_cxx11::array< std::vector< double >, 2 > get_coordinates () const;
+            void                                         get_data_from_file ();
+
+          private:
+
+            topo_types topo_type;
+            std::string topo_file;
+            std::vector<std::vector<std::vector<double> > > point_lists;
+            std::vector<double> topography_values;
+            // Gridded topography data
+            Function<2> *topography_data = NULL;
```

Comment:
In C++03, you can't do the initialization here. You have to do it in the constructor.